src/renderer/components/item-object-list/item-list-layout.tsx

```diff
@@ -197,6 +211,20 @@ export class ItemListLayout extends React.Component<ItemListLayoutProps> {
     return this.applyFilters(filterItems, allItems);
   }
 
+  updateColumnFilter(checkboxValue: boolean, className: string) {
+    if (checkboxValue){
+      this.columnProps.add(className);
+    } else {
+      this.columnProps.delete(className);
+    }
+  }
+
+  showColumn(index: number): boolean {
+    const {isConfigurable, tableId , renderTableHeader} = this.props;
+    if (!isConfigurable || !tableId) return true;
+    return this.columnProps.has(renderTableHeader[index].className);
```

Comment:
Are we assuming that `isConfigurable && tableId` ==> `renderTableHeader`? I think it would probably be a good idea if you added the following function.

```typescript
get canBeConfigured(): boolean {
  const { isConfigurable, tableId, renderTableHeader } = this.props;

  if (!isConfigurable || !tableId) {
    return false;
  }

  if (!renderTableHeader?.all(({ className }) => className)) {
    logger.warning("[ItemObjectList]: cannot configure an object list without all headers being identifiable");
    return false;
  }

  return true;
}
```